src/iarc7_motion/iarc_tasks/track_roomba_task.py

```diff
@@ -57,42 +60,54 @@ def __init__(self, task_dictionary):
             'roombas', OdometryArray, 
             self._receive_roomba_status)
 
+        self._current_velocity_sub = rospy.Subscriber(
+            '/odometry/filtered', Odometry,
+            self._current_velocity_callback)
+
         try:
             self._TRANSFORM_TIMEOUT = rospy.get_param('~transform_timeout')
+            self._TRACK_HEIGHT = rospy.get_param('~track_roomba_height')
             self._MAX_TRANSLATION_SPEED = rospy.get_param('~max_translation_speed')
             self._MAX_START_TASK_DIST = rospy.get_param('~roomba_max_start_task_dist')
             self._MAX_END_TASK_DIST = rospy.get_param('~roomba_max_end_task_dist')
+            self._MAX_Z_VELOCITY = rospy.get_param('~max_z_velocity')
             self._K_X = rospy.get_param('~k_term_tracking_x')
             self._K_Y = rospy.get_param('~k_term_tracking_y')
         except KeyError as e:
             rospy.logerr('Could not lookup a parameter for track roomba task')
             raise
 
         self._z_holder = HeightHolder()
+        self._height_checker = HeightChecker()
 
         self._state = TrackObjectTaskState.init
 
     def _receive_roomba_status(self, data):
         with self._lock:
             self._roomba_array = data
 
+    def _current_velocity_callback(self, data):
+        with self._lock:
+            self._drone_odometry = data
+
     def get_desired_command(self):
         with self._lock:
             if (self._state != TrackObjectTaskState.track):
-                if self._roomba_array is None:
+                if self._roomba_array is None or self._drone_odometry is None:
                     self._state = TrackObjectTaskState.waiting
                     return (TaskRunning(), NopCommand())
                 else:
                     self._state = TrackObjectTaskState.track
+            elif not (self._height_checker.above_min_maneuver_height(
+                        self._drone_odometry.pose.pose.position.z)):
+                return (TaskAborted(msg='Drone is too low'),)
+            elif not (self._height_checker.check_z_error(
```

Comment:
Seeing how you use this, I don't think this should be in the height checker. It's kinda specific to this task.